packages/editor/src/components/post-trash/check.js

```diff
@@ -34,10 +34,14 @@ export default function PostTrashCheck( { children } ) {
 			: false;
 
 		return {
-			canTrashPost: ( ! isNew || postId ) && canUserDelete,
+			canTrashPost:
+				( ! isNew || postId ) &&
+				canUserDelete &&
+				postType !== 'wp_template' &&
+				postType !== 'wp_block' &&
+				postType !== 'wp_template_part',
```

Comment:
I think `wp_navigation` should also be here. You could reuse `GLOBAL_POST_TYPES` from `store`.

https://github.com/WordPress/gutenberg/blob/57c59d1109f8c475eb2899dd7a4b1d5061a1df69/packages/editor/src/store/constants.ts#L32-L35